### ISSUE TITLE
Unescaped and case insensitive query parameters 

### ISSUE THREAD

**Issue by benbalter** [codeforamerica/cfapi#285]
When trying to query for Code For All organizations via the `http://codeforamerica.org/api/organizations` endpoint, most libraries will (properly) escape the `+`s in the expected query argument of `type=Code+For+All`, resulting in a query of `type=Code%2BFor%2B%All`.

The API should unescape the query arguments before querying, to allow for querying via properly escaped URLs (and thus clients can rely on libraries without having to manually build the query string).

Less critical, but it'd also be awesome if the query parameter were case insensitive (meaning `type=brigade` would return results as well as `type=Brigade` (what's currently expected)).

**Comment by ondrae** [comment]
Hi @benbalter.

We definitely want to start including case insensitive searching. Seeing a 404 for http://www.codeforamerica.org/brigade/code-for-san-francisco/ is no good.

I don't fully grok what you are asking for with escaping. I can run all the following without any issues
```
curl https://www.codeforamerica.org/api/projects?q=Code+for+San+Francisco
curl https://www.codeforamerica.org/api/projects?q=Code%2Bfor%2BSan%2BFrancisco
curl https://www.codeforamerica.org/api/projects?q=Code%20for%20San%20Francisco
curl https://www.codeforamerica.org/api/projects?q=Code+for%2BSan%20Francisco
```

Do you have an example where a library you are using is struggling?

**Comment by benbalter** [comment]
> Do you have an example where a library you are using is struggling?

Compare
 
`curl "http://codeforamerica.org/api/organizations?type=Code%2BFor%2BAll"`

with 

`curl "http://codeforamerica.org/api/organizations?type=Code+For+All"`

The first returns no results, the second returns 24 organizations.

**Comment by ondrae** [comment]
Ah, I see.  We are handling our search slightly different from the attribute matching.
https://github.com/codeforamerica/cfapi/blob/master/app.py#L199

We'll update all the places where we do string substitution and replace them with proper formatting.

Thanks for filing a ticket @benbalter

**Comment by tmaybe** [comment]
The unescaped parameters will will be fixed when #291 is merged and deployed. I can't reproduce the case-sensitive error as described above; these pairs of queries return identical results for me:

http://codeforamerica.org/api/organizations?type=Brigade
http://codeforamerica.org/api/organizations?type=brigade

http://codeforamerica.org/api/organizations?id=Code-for-San-Francisco
http://codeforamerica.org/api/organizations?id=code-for-san-francisco

**Comment by tmaybe** [comment]
Started a new issue for the case-sensitivity @ondrae described in #292

**Comment by tmaybe** [comment]
closed with the merging of #291
